### PR TITLE
Implement new `MapType` logical type

### PR DESCRIPTION
**Pull request by stloyd**
<h2>Change Log</h2>
<div id="change-log">
  <h4>Added</h4>
  <ul id="added">
    <li>Implement new `MapType` logical type</li>
  </ul> 
  <h4>Fixed</h4>  
  <ul id="fixed">
    
  </ul>
  <h4>Changed</h4>
  <ul id="changed">
    
  </ul>  
  <h4>Removed</h4>
  <ul id="removed">
    
  </ul>
  <h4>Deprecated</h4>
  <ul id="deprecated">
    
  </ul>  
  <h4>Security</h4> 
  <ul id="security">
    
  </ul>     
</div>
<hr/>

<h2>Description</h2>

### CODE REVIEW

File: src/core/etl/src/Flow/ETL/Row/Entry/TypedCollection.php [hunk 1]
```diff
@@ -2,9 +2,13 @@
...
+/**
+ * @template T
+ */
 interface TypedCollection
 {
-    public function type() : ListType;
+    /**
+     * @return T
+     */
+    public function type();
```
Comment (norberttech) [inline src/core/etl/src/Flow/ETL/Row/Entry/TypedCollection.php]: Why not `type(): Type` ?

File: src/core/etl/src/Flow/ETL/PHP/Type/Logical/MapType.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+use Flow\ETL\PHP\Type\Logical\Map\MapKey;
+use Flow\ETL\PHP\Type\Logical\Map\MapValue;
+use Flow\ETL\PHP\Type\Type;
+use Flow\Serializer\Serializable;
+
+/**
+ * @implements Serializable<array{key: MapKey, value: MapValue}>
+ */
+final class MapType implements LogicalType, Serializable
```
Comment (norberttech) [inline src/core/etl/src/Flow/ETL/PHP/Type/Logical/MapType.php:13]: We need to add unit tests for those types, there is not much logic here, but lack of coverage will reduce mutation score

Comment (stloyd) [reply src/core/etl/src/Flow/ETL/PHP/Type/Logical/MapType.php:13]: Covered, for other types I will make a new PR.

File: src/core/etl/src/Flow/ETL/Row/Entry/TypedCollection.php (hunk 1 above)
Comment (stloyd) [reply src/core/etl/src/Flow/ETL/Row/Entry/TypedCollection.php]: Reverted.

**Review by norberttech — APPROVED** [review]

**Pull request merged**